notebooks/concepts/specviz_glue_unit_conversion.ipynb

```diff
@@ -0,0 +1,421 @@
+    " - Should get data listen to display units?\n",
+    " - Feature branch or develop in place?\n",
+    "---\n",
+    "\n",
+    "## Acceptance criteria\n",
+    "\n",
+    " - Equivalencies list in notebook\n",
+    " - Make tickets\n",
+    " ---\n",
+    " \n",
+    "## Examples \n",
+    " Can make multiple unit_converter instances, which can handle different cases in Jdaviz\n",
+    " \n",
+    " - from glue.config import settings\n",
+    " - settings.UNIT_CONVERTER = 'custom-astropy'\n",
+    " - @unit_converter('custom-jdaviz')\n",
+    " - class UnitConverterWithSpectral:\n",
+    "\n",
+    "Using d1 below:\n",
+    "```\n",
+    "wcs1 = WCS(naxis=1)\n",
+    "wcs1.wcs.ctype = ['FREQ']\n",
+    "wcs1.wcs.crval = [1]\n",
+    "wcs1.wcs.cdelt = [1]\n",
+    "wcs1.wcs.crpix = [1]\n",
+    "wcs1.wcs.cunit = ['GHz']\n",
+    "\n",
+    "d1 = Data(f1=[1, 2, 3])\n",
+    "d1.get_component('f1').units = 'Jy'\n",
+    "d1.coords = wcs1\n",
+    "```\n",
+    "access with\n",
+    "```\n",
+    "astropy_converter = unit_converter.members[settings.UNIT_CONVERTER]()\n",
+    "equivalent_units = astropy_converter.equivalent_units(d1, d1.components[0], 'mJy')\n",
+    "converted_units = astropy_converter.to_unit(d1, 'Pixel Axis 0 [x]', d1.x, 'GHz', 'centimeter')\n",
+    "```\n",
+    "---\n",
+    "Dev notes\n",
+    "\n",
+    "y display unit is None and x display unit is set to 'Hz'\n",
+    "y axis label is Data values\n",
+    "x label is Frequency \\[Hz\\]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a6aa2289-d6f0-4f0e-969a-0e559f74d00a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import warnings\n",
+    "\n",
+    "import numpy as np\n",
+    "from astropy.wcs import WCS\n",
+    "import glue_jupyter as gj\n",
+    "from glue.core import Data\n",
+    "from glue.plugins.wcs_autolinking.wcs_autolinking import WCSLink\n",
+    "from specutils import Spectrum1D\n",
+    "from glue.core.link_helpers import LinkSame\n",
+    "from glue.core.roi import XRangeROI\n",
+    "from glue.config import settings\n",
+    "\n",
+    "\n",
+    "from jdaviz import Specviz"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "541bf389-6198-4d07-b744-4164d2ecfec4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def units_to_strings(unit_list):\n",
+    "    \"\"\"Convert equivalencies into readable versions of the units.\n",
+    "\n",
+    "    Parameters\n",
+    "    ----------\n",
+    "    unit_list : list\n",
+    "        List of either `astropy.units` or strings that can be converted\n",
+    "        to `astropy.units`.\n",
+    "\n",
+    "    Returns\n",
+    "    -------\n",
+    "    result : list\n",
+    "        A list of the units with their best (i.e., most readable) string version.\n",
+    "    \"\"\"\n",
+    "    all_units = []\n",
+    "    for unit in unit_list: \n",
+    "        try:\n",
+    "            u.Unit(unit)\n",
```

Comment:
You don't have to keep calling `u.Unit(unit)`. It has overhead. Save the result if successful and reuse it throughout the rest of the function.

```suggestion
    "            cur_unit = u.Unit(unit)\n",
```